crates/testing/src/spinning_task.rs

```diff
@@ -86,80 +79,120 @@ impl SpinningTaskDescription {
+                                                            "Node {} shutting down",
+                                                            idx
+                                                        );
+                                                        node.handle.shut_down().await;
+                                                    }
+                                                }
+                                                UpDown::NetworkUp => {
+                                                    if let Some(handle) = state.handles.get(idx) {
+                                                        tracing::error!(
+                                                            "Node {} networks resuming",
+                                                            idx
+                                                        );
+                                                        handle.networks.0.resume();
+                                                        handle.networks.1.resume();
+                                                    }
+                                                }
+                                                UpDown::NetworkDown => {
+                                                    if let Some(handle) = state.handles.get(idx) {
+                                                        tracing::error!(
+                                                            "Node {} networks pausing",
+                                                            idx
+                                                        );
+                                                        handle.networks.0.pause();
+                                                        handle.networks.1.pause();
+                                                    }
+                                                }
                                             }
                                         }
                                     }
+
+                                    // update our latest view
+                                    state.latest_view = Some(view_number);
                                 }
                             }
+
                             (None, state)
                         }
                         .boxed()
                     },
                 ));
+
+                let mut streams = vec![];
+                for handle in &mut state.handles {
+                    let s1 = handle
+                        .handle
+                        .get_event_stream_known_impl(FilterEvent::default())
+                        .await
+                        .0;
+                    let s2 = handle
+                        .handle
+                        .get_internal_event_stream_known_impl(FilterEvent::default())
+                        .await
+                        .0;
+                    streams.push(Merge::new(s1, s2));
+                }
```

Comment:
I'm a bit confused by this, we are merging the internal and external streams, but above it looks like we only use the external one?